src/frontend/src/catalog/root_catalog.rs

```diff
@@ -636,6 +636,7 @@ impl Catalog {
     }
 
     /// Used to get `TableCatalog` for Materialized Views, Tables and Indexes.
+    /// Retrieves all tables, created or creating.
     pub fn get_table_by_name<'a>(
```

Comment:
```suggestion
    pub fn get_any_table_by_name<'a>(
```

Can we make the name more explicit to prevent misuse? Also apply to other occurrences.

Reply:
Or, we just keep the original name and behavior but add a new function to also retrieve those being created. This is to reduce deviations from the convention and minimize changed lines.

Reply:
Prefer to rename to `get_any_table_by_name`.

Because the original `get_table_by_name` is really ambiguous, if we keep it.